lib/withReaderTaskEitherTimeout.ts

```diff
@@ -0,0 +1,56 @@
+import * as Either from 'fp-ts/Either';
+import { pipe } from 'fp-ts/function';
+import * as TaskEither from 'fp-ts/TaskEither';
+import { TimeoutError } from './errors.ts';
+import type * as ReaderTaskEither from 'fp-ts/ReaderTaskEither';
+
+const TIMEOUT_SYMBOL = Symbol('TimeoutReturnValue');
+
+type ReaderTaskEitherWithTimeoutCurriedFunc<
+  DependencyType,
+  LeftType,
+  RightType
+> = (
+  readerTaskEither: ReaderTaskEither.ReaderTaskEither<
+    DependencyType,
+    LeftType,
+    RightType
+  >
+) => ReaderTaskEither.ReaderTaskEither<
+  DependencyType,
+  TimeoutError | LeftType,
+  RightType
+>;
+
+export function withReaderTaskEitherTimeout<
+  DependencyType,
+  LeftType,
+  RightType
+>(
+  milliseconds: number
+): ReaderTaskEitherWithTimeoutCurriedFunc<DependencyType, LeftType, RightType> {
+  return (readerTaskEither) => {
+    return (deps: DependencyType) => {
+      const asyncTimeout = new Promise<typeof TIMEOUT_SYMBOL>((resolve) => {
+        setTimeout(() => resolve(TIMEOUT_SYMBOL), milliseconds);
```

Comment:
why don't you reject here and throw the error?

Reply:
1. I copy-pasted the code from an old implementation that Emil had put together
2. You cannot have a reject (error) type with promises, but you can have a resolve type. Therefore, I am using what the platform is giving me.

Reply:
but you dont care about the error type of the promise as you are casting it to an Either so you are explicitly defining the error. Anyway I just found it strange that you are resolving a symbol that you then transform to an Error